src/program/render.rs

```diff
@@ -0,0 +1,225 @@
+use crate::infobar;
+use crate::program::{make_dst, Program};
+use sdl2::image::LoadTexture;
+use sdl2::pixels::Color;
+use sdl2::rect::Rect;
+
+const PADDING: i32 = 30;
+const HALF_PAD: i32 = 15;
+
+impl<'a> Program<'a> {
+    /// render loads the image at the path in the images path vector located at the index and
+    /// renders to screen
+    pub fn render(&mut self) -> Result<(), String> {
+        self.screen.canvas.set_draw_color(dark_grey());
+        if self.paths.images.is_empty() {
+            return self.render_blank();
+        }
+        self.screen.canvas.clear();
+        self.render_image()?;
+        if self.ui_state.render_infobar {
+            self.render_infobar()?;
+        }
+
+        // Present to screen
+        self.screen.canvas.present();
+        Ok(())
+    }
+
+    fn render_image(&mut self) -> Result<(), String> {
+        self.set_image_texture()?;
+        match self.screen.last_texture {
+            Some(_) => (),
+            None => return Ok(()),
+        };
+        let tex = self.screen.last_texture.as_ref().unwrap();
+        let query = tex.query();
+        let target = self.screen.canvas.viewport();
+        let dest = make_dst(query.width, query.height, target.width(), target.height());
+        if let Err(e) = self.screen.canvas.copy(tex, None, dest) {
+            eprintln!("Failed to copy image to screen {}", e);
+        }
+        Ok(())
+    }
+
+    fn set_image_texture(&mut self) -> Result<(), String> {
+        if self.paths.index == self.screen.last_index
+            && self.screen.last_texture.is_some()
```

Comment:
This is just a way of rerendering an already loaded texture, instead of expensively reloading it before every render. It might not be the most elegant way to achieve this though, so I'm all ears for better ideas.